motionplan/utils.go

```diff
@@ -1,13 +1,57 @@
 package motionplan
 
 import (
+	"errors"
+	"fmt"
 	"math"
 
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/spatialmath"
 	"go.viam.com/rdk/utils"
 )
 
+// FrameStepsFromRobotPath is a helper function which will extract the waypoints of a single frame from the map output of a robot path.
+func FrameStepsFromRobotPath(frameName string, path []map[string][]referenceframe.Input) ([][]referenceframe.Input, error) {
+	solution := make([][]referenceframe.Input, 0, len(path))
+	for _, step := range path {
+		frameStep, ok := step[frameName]
+		if !ok {
+			return nil, fmt.Errorf("frame named %s not found in solved motion path", frameName)
```

Comment:
perhaps this should also use `NewFrameMissingError`. It realistically should never happen anyway

Reply:
Since this *should* never happen, I want to keep it a unique error, so if it *does* ever happen, it's clear where to look (and where not to)

Reply:
ok good with me